### PR TITLE
Feature/options menu

### PR DESCRIPTION
**Pull request by Leon910**

### CODE REVIEW

File: components/OptionsButton.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import styled from "styled-components";
+import OptionsIcon from "assets/optionsIcon/dots-circle-horizontal.svg";
+import { useState } from "react";
+import OptionsMenu from "./OptionsMenu";
+
+export default function OptionsButton({ onDeleteEmotion, emotion }) {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const handleClick = () => {
```
Comment (E-LeeSchaffer) [inline components/OptionsButton.js]: As the function is very simple and calls only one set state, I would suggest to pass the state directly in the jsx. Especially as it seems you only use the states here in this component.

File: components/OptionsButton.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    setIsMenuOpen(true);
+  };
+
+  const handleClosePopup = () => {
+    setIsMenuOpen(false);
+  };
+
+  return (
+    <>
+      <StyledButton onClick={handleClick}>
```
Comment (E-LeeSchaffer) [inline components/OptionsButton.js]:
```suggestion
      <StyledButton onClick={() => setIsMenuOpen(true)}>
```

File: components/OptionsButton.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <OptionsIcon />
+      </StyledButton>
+      {isMenuOpen && (
+        <OptionsMenu
+          onClose={handleClosePopup}
```
Comment (E-LeeSchaffer) [inline components/OptionsButton.js]:
```suggestion
          onClose={() => setIsMenuOpen(false)}
```

File: components/OptionsMenu.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import { useState } from "react";
+import Link from "next/link";
+
+export default function OptionsMenu({ onClose, onDeleteEmotion, emotion }) {
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  function toggleDeleteDialog() {
+    setIsDialogOpen(!isDialogOpen);
+  }
+
+  function handleDelete(emotion) {
```
Comment (E-LeeSchaffer) [inline components/OptionsMenu.js]: The emotion parameter is not used in this function. You can delete as you already have emotion from the props.
```suggestion
  function handleDelete() {
```

File: components/OptionsMenu.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    onDeleteEmotion(emotion._id);
+    toggleDeleteDialog();
+  }
+
+  console.log(emotion);
```
Comment (E-LeeSchaffer) [inline components/OptionsMenu.js]:
```suggestion
```

File: components/OptionsMenu.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+const StyledButtonCancel = styled.button`
+  margin: 10px;
+  padding: 10px 20px;
+  background-color: #95a5a6;
+  color: white;
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  &:hover {
```
Comment (E-LeeSchaffer) [inline components/OptionsMenu.js]: Actually you don't need a hover as this is a mobile view.

**Review by E-LeeSchaffer — COMMENTED** [review]
Hey, it looks good! I have only a few little comments.

**Comment by Leon910** [comment]
Thanks @E-LeeSchaffer I corrected the code due to your suggestions!

**Pull request merged**